tests/components/automation/test_state.py

```diff
@@ -423,3 +423,37 @@ def test_if_fires_on_entity_change_with_for(self):
         fire_time_changed(self.hass, dt_util.utcnow() + timedelta(seconds=10))
         self.hass.pool.block_till_done()
         self.assertEqual(1, len(self.calls))
+
+    def test_if_fires_on_for_condition(self):
+        self.hass.states.set('test.entity', 'on')
+        self.assertTrue(automation.setup(self.hass, {
+            automation.DOMAIN: {
+                'trigger': {
+                    'platform': 'event',
+                    'event_type': 'test_event',
+                },
+                'condition': {
+                    'platform': 'state',
+                    'entity_id': 'test.entity',
+                    'state': 'on',
+                    'for': {
+                        'seconds': 3
+                    },
+                },
+
+                'action': {
+                    'service': 'test.automation'
+                }
+            }
+        }))
+
+        # not enough time has passed
+        self.hass.bus.fire('test_event')
+        self.hass.pool.block_till_done()
+        self.assertEqual(0, len(self.calls))
+
+        # wait until we have passed the condition
+        time.sleep(4)
```

Comment:
Instead of the sleep I  think you can add
````
from tests.common import fire_time_changed
````
and then do
````
    fire_time_changed(self.hass, dt_util.utcnow() + timedelta(seconds=4))
    self.hass.pool.block_till_done()
````

At least that's what I stole from elsewhere!